### DISCUSSION TITLE
where to use Prisma middleware in Next js API route?

### DISCUSSION THREAD

**Discussion by moharifhidayat7** [prisma/prisma#12320] [Q&A] [answered]
i confused where to use prisma middleware in Next.js 12,
because Next.js 12 middleware is run on every request.

i have this lib/prisma.js file which i think where i should write prisma middleware. is it right?
```
import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

**Comment by austincrim** [comment, marked as answer]
Hey @moharifhidayat7 👋 ,

You can definitely add Prisma middleware here. I would add it with [`prisma.$use`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#use) before the default export at the end of the file.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
